src/ReactionService.Data/ReactionsGroupRepository.cs

```diff
@@ -16,6 +19,49 @@ public class ReactionsGroupRepository : IReactionsGroupRepository
     _provider = provider;
   }
 
+  private IQueryable<DbReactionsGroup> CreateGetPredicates(
+    GetReactionsGroupFilter filter)
+  {
+    IQueryable<DbReactionsGroup> query = _provider.ReactionsGroups.AsQueryable();
+
+    if (!filter.IncludeDeactivated)
+    {
+      query = query.Where(x => x.IsActive);
+    }
+
+    if (filter.IncludeReactions)
+    {
+      query = query.Include(x => x.Reactions.Where(x => x.IsActive));
+    }
+
+    return query;
+  }
+
+  public async Task<Guid?> CreateAsync(DbReactionsGroup dbReactionsGroup)
+  {
+    if (dbReactionsGroup is null)
+    {
+      return null;
+    }
+
+    _provider.ReactionsGroups.Add(dbReactionsGroup);
+    await _provider.SaveAsync();
+
+    return dbReactionsGroup.Id;
+  }
+
+  public Task<DbReactionsGroup> GetAsync(GetReactionsGroupFilter filter)
+  {
+    return filter is null
+      ? null
+      : CreateGetPredicates(filter).FirstOrDefaultAsync(x => x.Id == filter.ReactionsGroupId);
+  }
+
+  public Task<bool> DoesNameExist(string name)
+  {
+    return _provider.ReactionsGroups.AsNoTracking().AnyAsync(x => x.Name == name);
```

Comment:
Do we really need AsNoTracking() here?
Does Any() save anything in cache?

Reply:
No, it's not needed here. Done